### PR TITLE
feat(Cmake/Compiler): deprecated clang6 and gcc 7

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Set `clang` expected version `7.0.0`
- Set `GCC` expected version `8.0.0`

## Issues Addressed:
- Fix for #5611

### CODE REVIEW

File: .github/workflows/core_build.yml [hunk 1]
```diff
@@ -29,10 +29,6 @@ jobs:
             compiler: clang9
             modules: without
             extra_logs: false
-          - os: ubuntu-18.04
-            compiler: clang # default in 18.04 is clang 6
-            modules: without
-            extra_logs: false
```
Comment (FrancescoBorzi) [inline .github/workflows/core_build.yml:35]: please do not just remove the clang6 but **replace** it with clang7 instead

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]
also don't forget to update the wiki

**Comment by FrancescoBorzi** [comment]
we should also update all modules and replace clang 6 with clang 7

File: .github/workflows/core_build.yml (hunk 1 above)
Comment (Winfidonarleyan) [reply .github/workflows/core_build.yml:35]: Okay

Comment (Winfidonarleyan) [reply .github/workflows/core_build.yml:35]: Done

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan also don't forget: https://github.com/azerothcore/azerothcore-wotlk/blob/master/.github/SECURITY.md

**Comment by Winfidonarleyan** [comment]
@FrancescoBorzi done

**Comment by Winfidonarleyan** [comment]
modules, okay

**Comment by Winfidonarleyan** [comment]
Added modules PRs

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan failing with:

> Unknown compiler clang7

I'm ok if we remove clang6 and 7 from modules and only keep the default ubuntu 20.04 clang version in them. No need to test modules with different clang versions.

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan failing with:
> 
> > Unknown compiler clang7
> 
> I'm ok if we remove clang6 and 7 from modules and only keep the default ubuntu 20.04 clang version in them. No need to test modules with different clang versions.

Fixed

**Review by FrancescoBorzi — APPROVED** [review]

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**